lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-module/src/main/java/io/lighty/applications/rnc/module/RncLightyModule.java

```diff
@@ -171,22 +171,20 @@ private void startAndWaitLightyModule(final LightyModule lightyModule) throws Rn
     public boolean close() {
         LOG.info("Stopping RNC lighty.io application...");
         boolean success = true;
-        if (rncModuleConfig.getServerConfig().isEnableSwagger()
-                && this.swagger != null && !stopAndWaitLightyModule(this.swagger)) {
-            success = false;
+        if (rncModuleConfig.getServerConfig().isEnableSwagger() && this.swagger != null) {
+            success = swagger.shutdown(lightyModuleTimeout, DEFAULT_LIGHTY_MODULE_TIME_UNIT);
         }
-        if (this.rncModuleConfig.getAaaConfig().isEnableAAA()
-                && this.aaaLighty != null && !stopAndWaitLightyModule(this.aaaLighty)) {
-            success = false;
+        if (this.rncModuleConfig.getAaaConfig().isEnableAAA() && this.aaaLighty != null) {
+            success = aaaLighty.shutdown(lightyModuleTimeout, DEFAULT_LIGHTY_MODULE_TIME_UNIT);
         }
-        if (this.lightyRestconf != null && !stopAndWaitLightyModule(this.lightyRestconf)) {
-            success = false;
+        if (this.lightyRestconf != null) {
+            success = lightyRestconf.shutdown(lightyModuleTimeout, DEFAULT_LIGHTY_MODULE_TIME_UNIT);
```

Comment:
If previous shutdown fails and set `success` to false. This line can override the previous value to true. So at the end, success==true.

Reply:
Done